src/box/sql/util.c

```diff
@@ -1144,3 +1144,31 @@ sql_escaped_name_new(const char *name)
 	buf[size] = '\0';
 	return buf;
 }
+
+char *
+sql_legacy_name_new(const char *name, int len)
+{
+	size_t size = len + 1;
+	char *res = sql_xmalloc(size);
+	if (sqlIsquote(name[0])) {
+		memcpy(res, name, len);
+		res[len] = '\0';
+		sqlDequote(res);
+		return res;
+	}
+
+	UErrorCode status = U_ZERO_ERROR;
+	assert(icu_ucase_default_map != NULL);
+	int new_len = ucasemap_utf8ToUpper(icu_ucase_default_map, res, size,
+					   name, len, &status);
+	assert(U_SUCCESS(status) || status == U_BUFFER_OVERFLOW_ERROR);
+	if (new_len <= len)
+		return res;
+	size = new_len + 1;
+	res = sql_xrealloc(res, size);
+	new_len = ucasemap_utf8ToUpper(icu_ucase_default_map, res, size,
+				       name, len, &status);
+	assert((size_t)new_len < size);
```

Comment:
Minor: I'd rather recommend to use `if` + `unreachable` here. Feel free to ignore.

Reply:
I think it’s better to leave it as it is, because we already have such cases. Also, why `if` if better? The assertion will be removed from the release build, but the `if` will not be there unless optimized, right?

Reply:
AFAIU, `__buitin_unreachable` is used for compiler optimizations, but there is [`-funreachable-traps`](https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html#index-funreachable-traps) compiler flag, that forces to emit trap instruction into mcode (according to the docs). As a result, this might be used to check the assertion even in Release build, but I failed to reproduce this described behaviour in [godbolt](https://godbolt.org/z/h54Wxsjvq), so I doubt it worths playing with `unreachable` (at least for now).

Furthermore, I see this is not widely used in our sources (comparing to `assert`):
```bash
$ grep -rnF 'unreachable()' src/ | wc -l
204
$ grep -rnP '\s+assert\(.*\);' src/ | wc -l
4993
```

All in all, let's leave it as is, since there is no defined practice for `unreachable`.